common/persistence/sql/sqlShardPersistence.go

```diff
@@ -0,0 +1,322 @@
+	if err == nil {
+		// The shard already exists.
+		return &persistence.ShardAlreadyExistError{
+			Msg: fmt.Sprintf("CreateShard operation failed. Shard already exists. ShardId: %v, RangeId: %v", request.ShardInfo.ShardID, request.ShardInfo.RangeID),
+		}
+	} else {
+		switch err.(type) {
+		case *workflow.EntityNotExistsError:
+			row, err := shardInfoToShardsRow(*request.ShardInfo)
+			if err != nil {
+				return &workflow.InternalServiceError{
+					Message: fmt.Sprintf("CreateShard operation failed. Error: %v", err),
+				}
+			}
+
+			if _, err := m.db.NamedExec(createShardSQLQuery, row); err != nil {
+				return nil
+			}
+		default:
+			return &workflow.InternalServiceError{
+				Message: fmt.Sprintf("CreateShard operation failed. Failed to verify whether a shard of ID %v already existed. Error: %v", request.ShardInfo.ShardID, err),
+			}
+		}
+	}
+
+	return nil
+}
+
+func (m *sqlShardManager) GetShard(request *persistence.GetShardRequest) (*persistence.GetShardResponse, error) {
+	var row shardsRow
+	if err := m.db.Get(&row, getShardSQLQuery, request.ShardID); err != nil {
+		if err == sql.ErrNoRows {
+			return nil, &workflow.EntityNotExistsError{
+				Message: fmt.Sprintf("GetShard operation failed. Shard with ID %v not found. Error: %v", request.ShardID, err),
+			}
+		}
+		return nil, &workflow.InternalServiceError{
+			Message: fmt.Sprintf("GetShard operation failed. Failed to get record. ShardId: %v. Error: %v", request.ShardID, err),
+		}
+	}
+
+	clusterTransferAckLevel := make(map[string]int64)
+	if err := gobDeserialize(row.ClusterTransferAckLevel, &clusterTransferAckLevel); err != nil {
+		return nil, &workflow.InternalServiceError{
+			Message: fmt.Sprintf("GetShard operation failed. Failed to deserialize ShardInfo.ClusterTransferAckLevel. ShardId: %v. Error: %v", request.ShardID, err),
+		}
+	}
+
+	clusterTimerAckLevel := make(map[string]time.Time)
+	if err := gobDeserialize(row.ClusterTimerAckLevel, &clusterTimerAckLevel); err != nil {
+		return nil, &workflow.InternalServiceError{
+			Message: fmt.Sprintf("GetShard operation failed. Failed to deserialize ShardInfo.ClusterTimerAckLevel. ShardId: %v. Error: %v", request.ShardID, err),
+		}
+	}
+
+	return &persistence.GetShardResponse{ShardInfo: &persistence.ShardInfo{
+		ShardID:                   int(row.ShardID),
+		Owner:                     row.Owner,
+		RangeID:                   row.RangeID,
+		StolenSinceRenew:          int(row.StolenSinceRenew),
+		UpdatedAt:                 row.UpdatedAt,
+		ReplicationAckLevel:       row.ReplicationAckLevel,
+		TransferAckLevel:          row.TransferAckLevel,
+		TimerAckLevel:             row.TimerAckLevel,
+		ClusterTransferAckLevel:   clusterTransferAckLevel,
+		ClusterTimerAckLevel:      clusterTimerAckLevel,
+		DomainNotificationVersion: row.DomainNotificationVersion,
+	}}, nil
+}
+
+func (m *sqlShardManager) UpdateShard(request *persistence.UpdateShardRequest) error {
+	tx, err := m.db.Beginx()
```

Comment:
You don't need transaction/locking because there is only one sql statement to execute. It will be the same case for create.